src/viam/app/data_client.py

```diff
@@ -233,59 +255,72 @@ async def tabular_data_by_filter(
+        last: Optional[str] = None,
+        include_binary_data: bool = True,
+        count_only: bool = False,
+        include_internal_data: bool = False,
+        dest: Optional[str] = None,
+    ) -> Tuple[List[BinaryData], int, str]:
+        """Filter and download binary data. The data will be paginated into pages of `limit` items, and the pagination ID will be included
+        in the returned tuple. If a destination is provided, the data will be saved to that file.
+        If the file is not empty, it will be overwritten.
 
         ::
 
             from viam.proto.app.data import Filter
 
-            my_filter = Filter(component_type="camera")
-            binary_data = await data_client.binary_data_by_filter(my_filter)
 
-        Args:
-            filter (Optional[viam.proto.app.data.Filter]): Optional `Filter` specifying binary data to retrieve. No `Filter` implies all
-                binary data.
-            dest (Optional[str]): Optional filepath for writing retrieved data.
-            include_file_data (bool): Boolean specifying whether to actually include the binary file data with each retrieved file. Defaults
-                to true (i.e., both the files' data and metadata are returned).
-            num_files (Optional[str]): Number of binary data to return. Passing 0 returns all binary data matching the filter no matter.
-                Defaults to 100 if no binary data is requested, otherwise 10. All binary data or the first `num_files` will be returned,
-                whichever comes first.
+            my_data = []
+            last = None
+            my_filter = Filter(component_name="camera")
+            while True:
+                data, last = await data_client.binary_data_by_filter(my_filter, last)
+                if not data:
+                    break
+                my_data.extend(data)
 
-        Raises:
-            ValueError: If `num_files` is less than 0.
+        Args:
+            filter (viam.proto.app.data.Filter): Optional `Filter` specifying tabular data to retrieve. No `Filter` implies all binary
+                data.
+            limit (int): The maximum number of entries to include in a page. Defaults to 50.
+            sort_order (viam.proto.app.data.Order): The desired sort order of the data.
+            last (str): Optional string indicating the ID of the last-returned data.
+                        If provided, the server will return the next data entries after the `last` ID.
+            include_binary_data (bool): Boolean specifying whether to actually include the binary file data with each retrieved file.
+                                        Defaults to true (i.e., both the files' data and metadata are returned).
+            count_only (bool): Whether to return only the total count of entries.
+            include_internal_data (bool): Whether to return the internal data. Internal data is used for Viam-specific data ingestion,
```

Comment:
Since we're exposing `include_internal_data` exposed in the SDK, would great to add the corresponding [param](https://github.com/viamrobotics/api/blob/3e1d32e77cdb555adc41efc72a92259818b5cecb/proto/viam/app/data/v1/data.proto#L169) for TabularDataByFilter for consistency

Reply:
Should it be included?? I can remove it. I saw it in the proto so I added it